### PR TITLE
previously the auto-default to first option behavior for a required s…

### PR DESCRIPTION
**Pull request by boutell**
…elect field with no def setting only took effect visually, when you hit save you would be told the field was still blank.

### CODE REVIEW

File: modules/@apostrophecms/schema/ui/apos/components/AposInputSelect.vue [hunk 1]
```diff
@@ -44,6 +44,8 @@ export default {
   },
   data() {
     return {
+      next: (this.value && this.value.data != null)
```
Comment (boutell) [inline modules/@apostrophecms/schema/ui/apos/components/AposInputSelect.vue:47]: Overriding similar logic in the mixin.

**Review by abea — CHANGES_REQUESTED** [review]
I think the issue is more that it appears that the input has a value when it does not. Even if not required, someone could look at it, see a label, and assume that value was already chosen for them as the default (or previously selected).

**Comment by boutell** [comment]
The convention going back to A2 and before is that if a select element is required, the first option serves as the default value.

If we don't want that, then we need to have a design discussion about changing it. Whereas this PR fixes 3.x to behave as previous releases have.

So I would suggest that we accept this for now, as a bug fix, and that we open a ticket for a design discussion about changing the behavior if desired.

**Comment by abea** [comment]
That makes sense. As long as there is no null option, this seems fine. If there is a null option (not a required field) and has no configured default, it should not show any value and default to null, which I think is what we have already.

**Review by abea — DISMISSED** [review]

**Review by abea — DISMISSED** [review]

**Comment by boutell** [comment]
@abea I did dead code removal and purged the diff module here because it was blocking npm audit and thus the testing and the merge.

**Review by abea — APPROVED** [review]

**Pull request merged**